validator/src/network/rest/mod.rs

```diff
@@ -1,29 +1,59 @@
+use crate::services::mixmining;
 use iron::prelude::*;
+use presence::mixnode;
+use presence::topology;
 use router::Router;
+use std::sync::{Arc, Mutex};
 
-mod models;
-mod routes;
+mod capacity;
+mod presence;
+mod staking;
 
-pub struct Api {}
+pub struct Api {
+    mixmining_service: Arc<Mutex<mixmining::Service>>,
+}
 
 impl Api {
-    pub fn new() -> Api {
-        Api {}
+    pub fn new(mixmining_service: mixmining::Service) -> Api {
+        let service = Arc::new(Mutex::new(mixmining_service));
+        Api {
+            mixmining_service: service,
```

Comment:
Looking at how many places are using `Arc<Mutex<mixmining_service>>` perhaps we could make it into a type alias?